clusterloader2/pkg/config/codec.go

```diff
@@ -17,45 +17,35 @@ limitations under the License.
 package config
 
 import (
+	"bytes"
 	"fmt"
-	"io/ioutil"
-	"path/filepath"
-	"strings"
 
 	"github.com/spf13/viper"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/client-go/kubernetes/scheme"
 	"k8s.io/perf-tests/clusterloader2/api"
 )
 
-// ReadConfig creates test config from file specified by the given path.
-func ReadConfig(path string) (*api.Config, error) {
-	base := filepath.Base(path)
-	ext := filepath.Ext(base)
+// convertToConfig converts array of bytes into test config.
+// Config is parsed based on file extension. If not provided, yaml is assumed.
+func convertToConfig(raw []byte, ext string) (*api.Config, error) {
+	var config api.Config
 	v := viper.New()
-	v.SetConfigName(strings.TrimSuffix(base, ext))
-	v.AddConfigPath(filepath.Dir(path))
-	if err := v.ReadInConfig(); err != nil {
-		return nil, fmt.Errorf("read config failed: %v", err)
+	if ext == "" {
+		ext = "yaml"
+	}
+	v.SetConfigType(ext)
```

Comment:
Instead of relying on such fragile logic, I'd suggest re-using this utility - https://github.com/kubernetes/apimachinery/blob/master/pkg/util/yaml/decoder.go#L207

Reply:
Done.